staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -444,12 +439,6 @@ func (o *DebugOptions) debugByEphemeralContainer(ctx context.Context, pod *corev
 	pods := o.podClient.Pods(pod.Namespace)
 	result, err := pods.Patch(ctx, pod.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{}, "ephemeralcontainers")
 	if err != nil {
-		// The apiserver will return a 404 when the EphemeralContainers feature is disabled because the `/ephemeralcontainers` subresource
-		// is missing. Unlike the 404 returned by a missing pod, the status details will be empty.
-		if serr, ok := err.(*errors.StatusError); ok && serr.Status().Reason == metav1.StatusReasonNotFound && serr.ErrStatus.Details.Name == "" {
```

Comment:
I'd recommend keeping this code in place to give a helpful error on older servers